src/drive/web/modules/views/Recent/index.jsx

```diff
@@ -0,0 +1,144 @@
+    parentsDocsById = keyBy(parentsResult.data, '_id')
+  }
+
+  return {
+    ...result,
+    data: resultData.map(file => ({
+      ...file,
+      displayedPath: get(parentsDocsById[file.dir_id], 'path')
+    }))
+  }
+}
+
+const RecentView = ({ router, children }) => {
+  const { isBigThumbnail, toggleThumbnailSize } = useContext(
+    ThumbnailSizeContext
+  )
+
+  const query = buildRecentQuery()
+  const result = useFilesQueryWithPath(query)
+
+  const isInError = result.fetchStatus === 'error'
+  const hasDataToShow = !isInError && (result.data && result.data.length > 0)
+  const isLoading = !hasDataToShow && result.fetchStatus === 'loading'
+  const isEmpty = !isLoading && !hasDataToShow
+
+  const navigateToFolder = useCallback(folderId => {
+    router.push(`/folder/${folderId}`)
+  })
+
+  const navigateToFile = useCallback(file => {
+    router.push(`/recent/file/${file.id}`)
+  })
+
+  return (
+    <Main>
+      <ModalStack />
+      <Topbar>
+        <Breadcrumb />
+        <Toolbar canUpload={false} canCreateFolder={false} disabled={false} />
+      </Topbar>
+
+      <SelectionBar />
+      <FileListv2>
+        <MobileFileListHeader
+          folderId={null}
+          canSort={false}
+          thumbnailSizeBig={isBigThumbnail}
+          toggleThumbnailSize={toggleThumbnailSize}
+        />
+        <FileListHeader
+          folderId={null}
+          canSort={false}
+          thumbnailSizeBig={isBigThumbnail}
+          toggleThumbnailSize={toggleThumbnailSize}
+        />
+        <FileListBodyV2 selectionModeActive={false}>
+          {isInError && <Oops />}
+          {isLoading && <FileListRowsPlaceholder />}
+          {isEmpty && <EmptyDrive canUpload={false} />}
+          {hasDataToShow && (
+            <div className={isMobileApp() ? 'u-ov-hidden' : ''}>
+              {result.data.map(file => (
+                <File
+                  key={file._id}
+                  attributes={file}
+                  displayedFolder={null}
+                  onFolderOpen={navigateToFolder}
+                  onFileOpen={navigateToFile}
+                  withSelectionCheckbox={true}
+                  withFilePath={true}
+                  withSharedBadge={true}
+                  isFlatDomain={true}
+                  thumbnailSizeBig={isBigThumbnail}
+                />
+              ))}
+              {result.hasMore && <LoadMore fetchMore={result.fetchMore} />}
+            </div>
+          )}
+        </FileListBodyV2>
+      </FileListv2>
+      {children}
+    </Main>
+  )
+}
+
+const RecentViewWithProvider = props => (
+  <SharingProvider doctype="io.cozy.files" documentType="Files">
+    <ThumbnailSizeContextProvider>
+      <RouterContextProvider>
+        <ModalContextProvider>
+          <RecentView {...props} />
+        </ModalContextProvider>
+      </RouterContextProvider>
+    </ThumbnailSizeContextProvider>
+  </SharingProvider>
```

Comment:
It seems that this would be repeated between the Drive view and the Recent view, could we find a way to share both contexts ?

Reply:
I guess this will be changed in next PRs https://github.com/cozy/cozy-drive/pull/2041#discussion_r446099915